src/GraphQl/Subscription/SubscriptionManager.php

```diff
@@ -63,26 +73,129 @@ public function retrieveSubscriptionId(array $context, ?array $result, ?Operatio
+                unset($result['__private_field_'.$privateField]);
+            }
+        }
         $subscriptions[] = [$subscriptionId, $fields, $result];
         $subscriptionsCacheItem->set($subscriptions);
         $this->subscriptionsCache->save($subscriptionsCacheItem);
 
+        $this->updateSubscriptionCollectionCacheData(
+            $iri,
+            $fields,
+            $subscriptions,
+        );
+
         return $subscriptionId;
     }
 
-    public function getPushPayloads(object $object): array
+    public function getPushPayloads(object $object, string $type): array
+    {
+        if ('delete' === $type) {
+            $payloads =  $this->getDeletePushPayloads($object);
+        } else {
+            $payloads = $this->getCreatedOrUpdatedPayloads($object);
+        }
+
+        return $payloads;
+    }
+
+    /**
+     * @return array<array>
+     */
+    private function getSubscriptionsFromIri(string $iri): array
+    {
+        $subscriptionsCacheItem = $this->subscriptionsCache->getItem($this->encodeIriToCacheKey($iri));
+
+        if ($subscriptionsCacheItem->isHit()) {
+            return $subscriptionsCacheItem->get();
+        }
+
+        return [];
+    }
+
+    private function removeItemFromSubscriptionCache(string $iri): void
+    {
+        $cacheKey = $this->encodeIriToCacheKey($iri);
+        if ($this->subscriptionsCache->hasItem($cacheKey)) {
+            $this->subscriptionsCache->deleteItem($cacheKey);
+        }
+    }
+
+    private function encodeIriToCacheKey(string $iri): string
+    {
+        return str_replace('/', '_', $iri);
+    }
+
+    private function updateSubscriptionCollectionCacheData(
+        ?string                       $iri,
+        array                         $fields,
+        array                         $subscriptions,
+    ): void
+    {
+        $subscriptionCollectionCacheItem = $this->subscriptionsCache->getItem(
+            $this->encodeIriToCacheKey($this->getCollectionIri($iri)),
+        );
+        if ($subscriptionCollectionCacheItem->isHit()) {
+            $collectionSubscriptions = $subscriptionCollectionCacheItem->get();
+            foreach ($collectionSubscriptions as [$subscriptionId, $subscriptionFields, $subscriptionResult]) {
+                if ($subscriptionFields === $fields) {
+                    return;
+                }
+            }
+        }
+        $subscriptionCollectionCacheItem->set($subscriptions);
+        $this->subscriptionsCache->save($subscriptionCollectionCacheItem);
+    }
+
+    private function getResourceId(mixed $privateField, object $previousObject): string
+    {
+        $id = $previousObject->{'get' . ucfirst($privateField)}()->getId();
+        if ($id instanceof \Stringable) {
+            return (string)$id;
+        }
+        return $id;
+    }
+
+    private function getCollectionIri(string $iri): string
+    {
+        return substr($iri, 0, strrpos($iri, '/'));
+    }
```

Comment:
This is too hackish we need to find a better way.

Reply:
I'm all ears, I haven't found one :)

Reply:
you need to get the Collection Operation from a given resource: 
- get the resource class
- call the resource collection metadata factory
- call `getOperation` with forceCollection: true https://github.com/api-platform/core/blob/82ab5ef14d6041bd296db3b4ca917e01dd6ebae5/src/Metadata/Resource/ResourceMetadataCollection.php#L38

Then generate the IRI with the IRI Converter.